tcf_core/auth_pipeline.py

```diff
@@ -44,24 +45,50 @@ def password_validation(backend, details, request, response, *args, **kwargs):
             return redirect("/login/password_error", error=True)
     return {"password": response.get("password")}
 
+# Make BaseStrategy.validate_email() always return True, so that
+# social_core.pipeline.mail.mail_validation can be run multiple times
+BaseStrategy.validate_email = lambda *args, **kwargs: True
+
+def partial_token_persistence(orig_func):
+    """Wrapper over strategy.clean_partial_pipeline() to implement partial token persistence."""
+    @wraps(orig_func)
+    def new_func(self, *args, **kwargs):
+        if self.session.get('persist_partial_token', default=False):
+            # Persist partial token
+            pass
+        else:
+            # Delete partial token
+            orig_func(self, *args, **kwargs)
+    return new_func
+BaseStrategy.clean_partial_pipeline = partial_token_persistence(BaseStrategy.clean_partial_pipeline)
 
-@partial
```

Comment:
Instead of creating a new partial token for `collect_extra_info`, we'll persist and reuse the partial token from `social_core.pipeline.mail.mail_validation`.